library/src/main/java/com/getstream/sdk/chat/view/messageinput/AttachmentsController.kt

```diff
@@ -41,31 +39,27 @@ internal class AttachmentsController(
     internal fun onClickCloseAttachmentSelectionMenu() {
         view.hideAttachmentsMenu()
         totalAttachments = emptySet()
-        totalFileAttachmentAdapter.clear()
         totalMediaAttachmentAdapter.clear()
         configAttachmentButtonVisible(true)
     }
 
-    private fun fillTotalAttachmentsView(
-        messageInputType: MessageInputType?,
-        isMedia: Boolean,
-        treeUri: Uri? = null
-    ) {
+    private fun fillTotalMediaAttachmentsView(messageInputType: MessageInputType?) {
         GlobalScope.launch(dispatchersProvider.mainDispatcher) {
             view.showLoadingTotalAttachments(true)
-            totalAttachments = getAttachmentsFromLocal(isMedia, treeUri)
+            totalAttachments = withContext(dispatchersProvider.ioDispatcher) {
+                storageHelper.getMediaAttachments(view.context).toSet()
+            }
             if (totalAttachments.isEmpty()) {
                 view.showMessage(R.string.stream_no_media_error)
                 onClickCloseAttachmentSelectionMenu()
             } else {
-                setTotalAttachmentAdapters(
+                setTotalMediaAttachmentAdapter(
                     totalAttachments.toList(),
                     selectedAttachments.toList(),
-                    messageInputType,
-                    isMedia
+                    messageInputType
                 )
             }
-            setSelectedAttachmentAdapter(messageInputType, isMedia)
+            setSelectedAttachmentAdapter(messageInputType, true)
```

Comment:
Previously, there was one codebase with one logic for media and non-media files. That's why we had to pass isMedia flag, now there is not any reason to do that. I'd like to move out it completely. What if we divide method `setSelectedAttachmentAdapter` for two (media, non-media) and invoke it separately. I think it makes code cleaner and easier.
Also we can omit `showSelectedAttachments` and invoke view directly to show media non media attachments.

Reply:
agree